core/src/main/java/org/lflang/target/property/DockerProperty.java

```diff
@@ -45,20 +46,26 @@ public DockerOptions fromAst(Element node, MessageReporter reporter) {
       enabled = true;
       for (KeyValuePair entry : node.getKeyvalue().getPairs()) {
         DockerOption option = (DockerOption) DictionaryType.DOCKER_DICT.forName(entry.getName());
-        if (option != null && option.equals(DockerOption.FROM)) {
-          from = ASTUtils.elementToSingleString(entry.getValue());
-        }
-        if (option != null && option.equals(DockerOption.RTI_IMAGE)) {
-          rti = ASTUtils.elementToSingleString(entry.getValue());
+        switch (option) {
+          case FROM -> from = ASTUtils.elementToSingleString(entry.getValue());
+          case RUN -> run = ASTUtils.elementToSingleString(entry.getValue());
```

Comment:
I'm OK with adding this feature, but I'd like to draw attention to it as something to keep in mind if and when we later decide to consider a different approach to Dockerfile generation. We have already discussed that (e.g. having a user write a template for the docker file and supplying values to plug into their template), so this point should not be a blocker for merging this PR.

Reply:
Yeah, I agree. I was also thinking in those directions. As we get more experience with containerized deployment of LF programs it might get clearer how we best expose the right knobs to the user. This was added to allow using `FROM: ubuntu` and then also specify `RUN apt update && apt install cmake` which is needed.